src/main/java/edu/byu/cs/service/UserService.java

```diff
@@ -47,37 +45,36 @@ public static Collection<RepoUpdate> adminGetRepoHistory(String repoUrl, String
             }
         } catch (Exception e) {
             LOGGER.error("Error getting repo updates:", e);
-            throw new InternalServerException(e.getMessage(), e);
+            throw new InternalServerErrorResponse(e.getMessage());
         }
 
         return updates;
     }
 
-    private static void setRepoUrl(String studentNetId, String repoUrl, String adminNetId)
-            throws BadRequestException, InternalServerException, WordOfWisdomViolationException {
+    private static void setRepoUrl(String studentNetId, String repoUrl, String adminNetId) {
         try {
             if (!isValidRepoUrl(repoUrl)) {
-                throw new BadRequestException("Invalid Github Repo Url. Check if the link is valid and points directly to a Github Repo.");
+                throw new BadRequestResponse("Invalid Github Repo Url. Check if the link is valid and points directly to a Github Repo.");
             }
-        } catch (BadRequestException e) {
+        } catch (BadRequestResponse e) {
             throw e;
         } catch (Exception e) {
             LOGGER.error("Error cloning repo during repoPatch: {}", e.getMessage());
-            throw new InternalServerException("There was an internal server error in verifying the Github Repo", e);
+            throw new InternalServerErrorResponse("There was an internal server error in verifying the Github Repo");
         }
 
         RepoUpdate historicalUpdate;
         try {
             historicalUpdate = verifyRepoIsAvailableForUser(repoUrl, studentNetId);
         } catch (DataAccessException e) {
-            throw new InternalServerException("There was an internal server error in checking the repo update history for this url", e);
+            throw new InternalServerErrorResponse("There was an internal server error in checking the repo update history for this url");
         }
         if (historicalUpdate != null) {
             if (adminNetId != null) {
-                throw new WordOfWisdomViolationException("Repo is blocked because of a prior claim: " + historicalUpdate);
+                throw new ImATeapotResponse("Repo is blocked because of a prior claim: " + historicalUpdate);
             } else {
                 LOGGER.info("Student {} was blocked from updating their url because of a prior claim: {}", studentNetId, historicalUpdate);
-                throw new WordOfWisdomViolationException("Please talk to a TA to submit this url");
```

Comment:
Kinda sad to see this hilariously but appropriately named Exception go, but probably for the best. Do we want to keep the 418 status though?